Rubberduck.VBEEditor/ComManagement/TypeLibs/TypeLibsAPI.cs

```diff
@@ -0,0 +1,1006 @@
+        {
+            using (var typeLib = TypeLibWrapper.FromVBProject(project))
+            {
+                return CompileComponent(typeLib, componentName);
+            }
+        }
+
+        /// <summary>
+        /// Compile a single VBA component (e.g. module/class)
+        /// </summary>
+        /// <remarks>NOTE: This will only return success if ALL components that this component depends on also compile successfully</remarks>
+        /// <param name="projectTypeLib">Low-level ITypeLib wrapper representing the VBA project</param>
+        /// <param name="componentName">The name of the component (module/class etc) to compile</param>
+        /// <returns>bool indicating success/failure.</returns>
+        public static bool CompileComponent(TypeLibWrapper projectTypeLib, string componentName)
+        {
+            return CompileComponent(projectTypeLib.TypeInfos.Get(componentName));
+        }
+
+        /// <summary>
+        /// Compile a single VBA component (e.g. module/class)
+        /// </summary>
+        /// <remarks>NOTE: This will only return success if ALL components that this component depends on also compile successfully</remarks>
+        /// <param name="component">Safe-com wrapper representing the VBA component to compile</param>
+        /// <returns>bool indicating success/failure.</returns>
+        public static bool CompileComponent(IVBComponent component)
+        {
+            return CompileComponent(component.ParentProject, component.Name);
+        }
+
+        /// <summary>
+        /// Compile a single VBA component (e.g. module/class)
+        /// </summary>
+        /// <remarks>NOTE: This will only return success if ALL components that this component depends on also compile successfully</remarks>
+        /// <param name="componentTypeInfo">Low-level ITypeInfo wrapper representing the VBA component to compile</param>
+        /// <returns>bool indicating success/failure.</returns>
+        public static bool CompileComponent(TypeInfoWrapper componentTypeInfo)
+        {
+            return componentTypeInfo.CompileComponent();
+        }
+
+        /// <summary>
+        /// Execute a routine inside a standard VBA code module
+        /// </summary>
+        /// <remarks>the VBA return value returned here can be a COM object, but needs freeing with Marshal.ReleaseComObject to ensure deterministic behaviour.</remarks>
+        /// <param name="ide">Safe-com wrapper representing the VBE</param>
+        /// <param name="projectName">VBA Project name, as declared in the VBE</param>
+        /// <param name="standardModuleName">Module name, as declared in the VBA project</param>
+        /// <param name="procName">Procedure name, as declared in the VBA module</param>
+        /// <param name="args">optional array of arguments to pass to the VBA routine</param>
+        /// <returns>object representing the VBA return value, if one was provided, or null otherwise.</returns>
```

Comment:
Should this note that we are responsible for releasing the returned object? What about primitive types (which may be returned as a variant and thus needs releasing nonetheless?)

If update is needed, note to update the XML-Docs for other overloads as well.

Reply:
there is an XML doc remark about returning COM objects.  There is no special case needed for other variants.